### ISSUE TITLE
To-dos before benchmarking a large DNN

### ISSUE THREAD

**Issue by jacobgao** [dimitrijejankov/simsql#6]
- [x] Fixing the memory issues of the 3 matrix-related UDFs (finally!)
- [ ] Mini-batch gradient descent (the most important one)
- [ ] ReLU and cross entropy loss function (to be consistent with TensorFlow)
- [ ] Preprocessing the bag-of-words Wikipedia data I provide so that SimSQL can load it in (also please consider how to pad all the 0's for every mini-batch or even use sparse matrix representation)
- [ ] Automating evaluation over a portion of the training set every hour (might be trivial - just need queries to output the cross entropy for a larger batch)

Meanwhile, I'll find a way to export the initial values for all the variables (weights, biases) in TensorFlow before we are ready to kick off the real comparison.
